exec/java-exec/src/main/resources/rest/profile/list.ftl

```diff
@@ -37,7 +37,15 @@
       <strong>No running queries.</strong>
     </div>
   </#if>
-  <h3>Completed Queries</h3>
+  <table width="100%">
+    <tr>
+      <td><h3>Completed Queries</h3></td>
+      <td align="right">
+        <form action="/profiles" method="get"><span title="Max number of profiles to list">Show </span>
+        <input id="fetchMax" type="text" size="7" name="max" value="${model.getFinishedQueries()?size}">
+        <input type="submit" value="Go">
+      </form></td>
+    </tr></table>
```

Comment:
BTW: we should validate the max parameter in the Java code. Force max into the range: 1 ... 500 (say).

That is, don't allow -1, 0 or 1000000 profiles.